devices/giex.js

```diff
@@ -1,135 +1,203 @@
-        cluster: 'manuSpecificTuya',
-        type: ['commandDataResponse', 'commandDataReport'],
-        convert: (model, msg, publish, options, meta) => {
-            for (const dpValue of msg.data.dpValues) {
-                const value = tuya.getDataValue(dpValue);
-                const dp = dpValue.dp;
-                switch (dp) {
-                case dataPoints.giexWaterValveState: {
-                    return {state: value ? 'ON': 'OFF'};
-                }
-                case dataPoints.giexWaterValveMode: {
-                    return {mode: value ? 'capacity': 'duration'};
-                }
-                case dataPoints.giexWaterValveIrrigationTarget: {
-                    return {irrigation_target: value};
-                }
-                case dataPoints.giexWaterValveCycleIrrigationNumTimes: {
-                    return {cycle_irrigation_num_times: value};
-                }
-                case dataPoints.giexWaterValveCycleIrrigationInterval: {
-                    return {cycle_irrigation_interval: value};
+    giexWaterValve: (timezone) =>
+        ({
+            cluster: 'manuSpecificTuya',
+            type: ['commandDataResponse', 'commandDataReport'],
+            convert: (model, msg, publish, options, meta) => {
+                for (const dpValue of msg.data.dpValues) {
+                    const value = tuya.getDataValue(dpValue);
+                    const dp = dpValue.dp;
+                    switch (dp) {
+                    case dataPoints.giexWaterValve.state:
+                        return {[keys.giexWaterValve.state]: value ? ON: OFF};
+                    case dataPoints.giexWaterValve.mode:
+                        return {[keys.giexWaterValve.mode]: value ? CAPACITY: DURATION};
+                    case dataPoints.giexWaterValve.irrigationTarget:
+                        return {[keys.giexWaterValve.irrigationTarget]: value};
+                    case dataPoints.giexWaterValve.cycleIrrigationNumTimes:
+                        return {[keys.giexWaterValve.cycleIrrigationNumTimes]: value};
+                    case dataPoints.giexWaterValve.cycleIrrigationInterval:
+                        return {[keys.giexWaterValve.cycleIrrigationInterval]: value};
+                    case dataPoints.giexWaterValve.waterConsumed:
+                        return {[keys.giexWaterValve.waterConsumed]: value};
+                    case dataPoints.giexWaterValve.irrigationStartTime:
+                        return {[keys.giexWaterValve.irrigationStartTime]: timezone ? toLocalTime(value, timezone) : value};
+                    case dataPoints.giexWaterValve.irrigationEndTime:
+                        return {[keys.giexWaterValve.irrigationEndTime]: timezone ? toLocalTime(value, timezone) : value};
+                    case dataPoints.giexWaterValve.lastIrrigationDuration:
+                        return {[keys.giexWaterValve.lastIrrigationDuration]: value.split(',').shift()}; // Remove meaningless ,0 suffix
```

Comment:
Note: for `_TZE200_a7sghmms` I've always seen `,0` (i.e. `00:02:10,0` instead of a proper fraction of a second, that's why I'm removing this.